cmd/aro/deploy.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/deploy"
+)
+
+func deployer(ctx context.Context, log *logrus.Entry) error {
+	for _, key := range []string{
+		"AZURE_TENANT_ID",
+		"AZURE_CLIENT_SECRET",
+		"AZURE_CLIENT_ID",
+		"AZURE_SUBSCRIPTION_ID",
+		"AZURE_RP_PARAMETERS_FILE",
+		"AZURE_RP_RESOURCEGROUP_NAME",
+	} {
+		if _, found := os.LookupEnv(key); !found {
+			return fmt.Errorf("environment variable %q unset", key)
+		}
+	}
+
+	if len(os.Getenv("AZURE_RP_VERSION")) > 0 {
+		gitCommit = os.Getenv("AZURE_RP_VERSION")
+	} else if gitCommit == "unknown" ||
+		strings.Contains(gitCommit, "dirty") {
+		return fmt.Errorf("gitCommit '%s' is not valid deployment version", gitCommit)
+	}
+
+	authorizer, err := auth.NewClientCredentialsConfig(os.Getenv("AZURE_CLIENT_ID"), os.Getenv("AZURE_CLIENT_SECRET"), os.Getenv("AZURE_TENANT_ID")).Authorizer()
```

Comment:
use auth.NewAuthorizerFromEnvironment.  You never know when we might end up using certificate or MSI-based authentication.

Reply:
ack